services/gateway/src/controller/order/order.controller.ts

```diff
@@ -398,7 +376,35 @@ orderRoutes.put("/api/order/claim/:id", async (req: Request, res: Response) => {
     });
   });
 
-  if (!(await check(token, { id: order.user?.id }))) return res.status(401).send({ error: "Unauthorized" });
+  const userRole = user.getRole()?.getCode();
+
+  if (userId !== order.user?.id && userRole !== "DELIVERY_PERSON")
+    return res.status(401).send({ error: "Unauthorized: user is not the delivery person nor the order owner" });
+
+  if (userRole === "DELIVERY_PERSON") {
```

Comment:
```suggestion
  // TODO: check if user is deliveryPerson of this order, to avoid if customer is also delivery_person
  if (userRole === "DELIVERY_PERSON") {
```